src/Particular.LicensingComponent/ThroughputCollector.cs

```diff
@@ -96,6 +96,15 @@ public async Task<ReportGenerationState> GetReportGenerationState(CancellationTo
             ReportCanBeGenerated = await dataStore.IsThereThroughputForLastXDays(30, cancellationToken)
         };
 
+        //ensure that if there is a broker that we only allow generation if there is broker data in the last 30 days
+        if (reportGenerationState.ReportCanBeGenerated && throughputQuery != null)
+        {
+            if (!await dataStore.IsThereThroughputForLastXDaysForSource(30, ThroughputSource.Broker, false, cancellationToken))
+            {
+                reportGenerationState.ReportCanBeGenerated = false;
+            }
+        }
+
         if (!reportGenerationState.ReportCanBeGenerated)
         {
             reportGenerationState.Reason = "24 hours worth of data needs to exist in the last 30 days.";
```

Comment:
I think we need to modify the reason based on what the issue actually is.  

Otherwise situations can arise where the customer sees that there is recent throughput data, but they're still not able to generate a report because it isn't broker data. And the reason they're given is that there is no recent data.

Reply:
@abparticular  There's limited space on the screen - we can't really add any more text than that, If you have a suggestion that doesn't change the length please give one. 
They would also be seeing a big red message on the screen saying that there are connection issues to the broker pointing them to fix it. So it should be pretty obvious in general what the issue is.